### PR TITLE
draft 3876

### PR DESCRIPTION
**Pull request by jleclanche**
Fix for #3876. Also lets us set `POOTLE_GIT_PATH` environment variable to let us override the path to the git binary.

### CODE REVIEW

**Comment by jleclanche** [comment]
@iafan does this resolve your issue?

File: pootle/core/utils/version.py [hunk 1]
```diff
@@ -183,6 +183,29 @@ def _shell_command(command):
...
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def _git_binary():
+    """
+    Tries to find the path to the `git` program.
+    Can be overridden by setting POOTLE_GIT_PATH environment variable.
+    See #3876.
+    """
+    setting = os.environ.get("POOTLE_GIT_PATH")
```
Comment (unho) [inline pootle/core/utils/version.py]: Is this really necessary to add a new custom `POOTLE_GIT_PATH` envvar? IMHO it is good enough to use the logic below.

**Comment by unho** [comment]
Please use a better commit message, one that includes a reference to the issue it tries to fix.

File: pootle/core/utils/version.py (hunk 1 above)
Comment (jleclanche) [inline pootle/core/utils/version.py]: It's free, honestly.

**Comment by iafan** [comment]
> @iafan does this resolve your issue?

I still get this traceback:

```
Traceback (most recent call last):
  File "manage.py", line 15, in <module>
    from pootle import syspath_override
  File "C:\Storage\Work\GitHub\pootle\pootle\__init__.py", line 14, in <module>
    __version__ = get_version(VERSION)
  File "C:\Storage\Work\GitHub\pootle\pootle\core\utils\version.py", line 64, in get_version
    git_changeset = get_git_changeset()
  File "C:\Program Files (x86)\Python\lib\site-packages\django\utils\lru_cache.py", line 125, in wrapper
    result = user_function(*args, **kwds)
  File "C:\Storage\Work\GitHub\pootle\pootle\core\utils\version.py", line 221, in get_git_changeset
    [_git_binary(), 'log', '--pretty=format:%ct', '--quiet', '-1', 'HEAD']
  File "C:\Storage\Work\GitHub\pootle\pootle\core\utils\version.py", line 180, in _shell_command
    universal_newlines=True
  File "C:\Program Files (x86)\Python\lib\subprocess.py", line 709, in __init__
    errread, errwrite)
  File "C:\Program Files (x86)\Python\lib\subprocess.py", line 957, in _execute_child
    startupinfo)
WindowsError: [Error 2]
```

**Comment by iafan** [comment]
@jleclanche, and the reason your code doesn't work on Windows is simply because on windows the executable file name is `git.exe`, not `git`.

I think we shouldn't be checking for the file existence for a dozen of times each time we try to get the git executable path. This is a more simple code that works:

```Python
def _git_binary():
    """
    Tries to find the path to the `git` program.
    Can be overridden by setting POOTLE_GIT_PATH environment variable.
    See #3876.
    """
    setting = os.environ.get("POOTLE_GIT_PATH")
    if setting:
        return setting

    trypath = "/usr/bin/git"
    if _is_executable_file(trypath):
        return trypath

    return "git"
```

**Pull request closed without merging**